### PR TITLE
D8 core 2127 & 2175: Merge in stanford starter theme updates, and added style basics

### PR DESCRIPTION
**Pull request by katrialesser**
PR Ready for Review, Will continue more styles after this is merged. 

Sorry I didn't merge the stanford theme starter updates in a separate branch, that probably would have been cleaner to read.

- Updated the theme from stanford_starter, checked differences etc.
- Added everything in the `src/scss/utilities folder` to have a base for color variables, and font & background mixins.
- Added some font styles in `src/scss/base` 
- Added some background styles in `src/scss/theme`

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
-    "name": "su-soe/soe_basic",
-    "description": "SOE Basic Branding Sub-Theme.",
-    "type": "drupal-custom-theme",
-    "homepage": "https://github.com/SU-SOE/soe_basic",
-    "authors": [],
-    "license": "GPL-2.0+",
-    "minimum-stability": "dev"
+  "name": "su-soe/soe_basic",
```
Comment (katrialesser) [inline composer.json:2]: no change, just indented 2 spaces instead of 4.

File: webpack.config.js [hunk 2]
```diff
@@ -43,40 +43,31 @@ const distAssets = path.resolve(__dirname, process.env.npm_package_config_distAs
...
+    path: path.resolve(distJS)
   },
   // Additional module rules.
   module: {
     rules: [
       // Drupal behaviors need special handling with webpack.
       // https://www.npmjs.com/package/drupal-behaviors-loader
       {
-        test: /\.behavior.js$/,
+        // test: /\.behavior.js$/,
```
Comment (sherakama) [inline webpack.config.js:70]: @katrialesser Why is this commented out? This is needed for the drupal behavior fixes FWIK.

File: src/scss/utilities/mixins/typography/index.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+//@charset 'UTF-8';
```
Comment (sherakama) [inline src/scss/utilities/mixins/typography/index.scss:1]: charset should be at the top of each sass file.

File: src/scss/utilities/variables/_colors.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/* SOE Specific colors, Stanford/Decanter colors are not defined here */
```
Comment (sherakama) [inline src/scss/utilities/variables/_colors.scss:1]: `@charset 'UTF-8';` should be at the top of each sass file.

File: src/scss/utilities/mixins/typography/_soe-sans-mixins.scss [hunk 5]
```diff
@@ -0,0 +1,46 @@
+
```
Comment (sherakama) [inline src/scss/utilities/mixins/typography/_soe-sans-mixins.scss:1]: `@charset 'UTF-8';` should be at the top of each sass file.

File: src/scss/utilities/mixins/typography/_soe-sans-mixins.scss [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+@mixin soe-sans {
+  font-family: $soe-font--sans;
+}
+
+
+// Note that main difference with basic styles is font-weight
+@mixin soe-type-sans-base {
+  font-family: $soe-font--sans; // should already be, but we can make sure
+  font-weight: 900;
+  //color: $su-color-black; // kerri do we need this?
```
Comment (sherakama) [inline src/scss/utilities/mixins/typography/_soe-sans-mixins.scss:13]: @kerri-augenstein

**Comment by sherakama** [comment]
@katrialesser Thanks for putting in this PR. I have enabled codeclimate and it picked up a few formatting items. I will leave a few additional comments as well.

File: src/scss/utilities/mixins/_gradients.scss [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+@mixin gradient--transparent-purple-lagunita {
+  background: linear-gradient(91.01deg, rgba($su-color-bright-purple, 0.9) 3.47%, rgba($su-color-lagunita, 0.8) 95.48%);
+}
+
+@mixin gradient--transparent-purple-green {
+  background: linear-gradient(89.49deg, rgba($su-color-bright-purple, 0.9) 0.24%, rgba($su-color-bright-palo-alto, 0.8) 102.73%);
+}
+
+@mixin gradient--transparent-purple-red {
+  //background: linear-gradient(270.22deg, #620059 0.12%, rgba(177, 4, 14, 0.8) 99.89%);
```
Comment (sherakama) [inline src/scss/utilities/mixins/_gradients.scss:21]: please remove commented out code.

File: src/scss/theme/index.scss [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+//
+// Purely visual styling, such as border, box-shadow, colors and backgrounds,
+// font properties, etc. Ideally, these should be separated enough from a
+// component’s structure to be “swappable”, and omitting these entirely should
+// not break the component’s functionality or basic usability.
+
+// Config and supporting scss.
+@import '../config/index';
+@import "../utilities/index";
+//@import 'node_modules/decanter/core/src/scss/decanter';
```
Comment (sherakama) [inline src/scss/theme/index.scss:17]: please remove commented out code.

File: soe_basic.info.yml [hunk 9]
```diff
@@ -29,6 +29,7 @@ component-libraries:
 
 libraries:
   - stanford_basic/basic
```
Comment (sherakama) [inline soe_basic.info.yml:31]: Do you need to declare the parent library here? Try removing it and see what happens.

File: src/scss/utilities/mixins/typography/index.scss (hunk 3 above)
Comment (katrialesser) [reply src/scss/utilities/mixins/typography/index.scss]: You're correct, my php storm was flagging all of them so i had tried commenting them out when the npm build wasn't building, sorry I need to change all of them back.

File: webpack.config.js (hunk 2 above)
Comment (katrialesser) [reply webpack.config.js:70]: Oh sorry! That was when I was debugging why the build wasn't working, changing back!

File: src/scss/utilities/mixins/typography/_soe-sans-mixins.scss (hunk 6 above)
Comment (katrialesser) [reply src/scss/utilities/mixins/typography/_soe-sans-mixins.scss:14]: Pinging Kerri in slack about this question.

File: src/scss/utilities/mixins/_gradients.scss (hunk 7 above)
Comment (katrialesser) [reply src/scss/utilities/mixins/_gradients.scss:22]: will do! thanks

File: soe_basic.info.yml (hunk 9 above)
Comment (katrialesser) [reply soe_basic.info.yml:31]: I also wondered that. I think it's old code leftover and we can remove it. It compiles fine without it.

File: src/scss/utilities/mixins/typography/_soe-sans-mixins.scss (hunk 6 above)
Comment (katrialesser) [reply src/scss/utilities/mixins/typography/_soe-sans-mixins.scss:14]: Resolved, removed.

**Pull request merged**